src/scripts/main.js

```diff
@@ -1,3 +1,59 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const headers = table.querySelectorAll('thead th');
+const rows = table.querySelectorAll('tbody tr');
+const tbody = table.querySelector('tbody');
+
+headers[0].setAttribute('data-type', 'text');
+headers[1].setAttribute('data-type', 'text');
+headers[2].setAttribute('data-type', 'age');
+headers[3].setAttribute('data-type', 'salary');
+
+const transform = (index, content) => {
+  const type = headers[index].getAttribute('data-type');
+
+  switch (type) {
+    case 'age':
+      return parseFloat(content);
+    case 'salary':
+      return parseFloat(content.slice(1).replaceAll(',', ''));
+    case 'text':
+      return content.toLowerCase().replaceAll(' ', '');
+  }
+};
+
+const sortColumn = (index) => {
+  const newRows = Array.from(rows);
+
+  newRows.sort((rowA, rowB) => {
+    const cellA = rowA.querySelectorAll('td')[index].innerHTML;
+    const cellB = rowB.querySelectorAll('td')[index].innerHTML;
+
+    const a = transform(index, cellA);
+    const b = transform(index, cellB);
+
+    switch (true) {
+      case a > b:
+        return 1;
+      case a < b:
+        return -1;
+      case a === b:
+        return 0;
+    }
+  });
+
+  [].forEach.call(rows, (row) => {
+    tbody.removeChild(row);
+  });
```

Comment:
couldn't we just iterate through rows?
https://stackoverflow.com/questions/16053357/what-does-foreach-call-do-in-javascript
read `Update for ES6(ES2015) and Beyond`

Reply:
https://attacomsian.com/blog/javascript-convert-nodelist-to-array#:~:text=A%20NodeList%20is%20an%20array,()%20on%20a%20NodeList%20object.

Reply:
better use spread